File: 24.cpp
//
// Created by zb198 on 2019/10/2.
//
struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(nullptr) {
	}
};
class Solution {
public:
    ListNode* ReverseList(ListNode* pHead) {
        if(pHead== nullptr){
            return nullptr;
        }
        ListNode* pNode = pHead->next;
        pHead->next = nullptr;
        while (pNode!= nullptr){
            ListNode* pTemp = pNode->next;
            pNode->next = pHead;
            pHead = pNode;
            pNode = pTemp;
            pTemp = nullptr;
        }
        return pHead;
    }
};

File: 24.py
# -*- coding:utf-8 -*-
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next:ListNode = None
class Solution:
    # 返回ListNode
    def ReverseList(self, pHead:ListNode)->ListNode:
        if(pHead == None):
            return None
        pNode = pHead.next
        pHead.next = None
        while(pNode!=None):
            pTemp = pNode.next
            pNode.next = pHead
            pHead = pNode
            pNode = pTemp
        return pHead


# write code here